gateway/proxy_fil.go

```diff
@@ -441,13 +441,23 @@ func (gw *Node) GetActorEvents(ctx context.Context, filter *types.ActorEventFilt
 	if err := gw.limit(ctx, stateRateLimitTokens); err != nil {
 		return nil, err
 	}
+	if filter != nil && filter.FromHeight != nil {
+		if err := gw.checkTipSetHeight(ctx, *filter.FromHeight, types.EmptyTSK); err != nil {
```

Comment:
Do we also need to apply some default "max events" limit? Maybe the lookback is enough?

Reply:
no, we have the `MaxFilterHeightRange` and `MaxFilterResults` in config that defaults to 2880 and 10000 respectively, which is checked wherever you supply a filter (both here and the eth forms)

(aside: the `MaxFilterResults` is pretty annoying because it treats the collected events as a FIFO with that as the limit, and the user doesn't get told that they reached the limit, it just silently bumps out previously found events to stay within the limit, we need to fix that I think. Notes about that included in https://github.com/filecoin-project/lotus/issues/11680)